### PR TITLE
Add vineyardctl for managing vineyard operator and relavant compenents on kubernetes

### PR DESCRIPTION
**Pull request by dashanji**
What do these changes do?
-------------------------

- Add vineyardoperatorctl(CLI tool) for deploying cert-manager and vineyard 
operator.
- Add relavant e2e test.

Related https://github.com/v6d-io/v6d/issues/1152

### CODE REVIEW

**Comment by dashanji** [comment]
@sighingnow Big big thanks for fixing the deps issue. ❤️❤️❤️

File: docker/Dockerfile.vineyard-python-dev [hunk 1]
```diff
@@ -15,7 +15,7 @@
 # build vineyard-python-dev
 FROM vineyardcloudnative/vineyard-manylinux1:20220512 as wheel
 
-ENV python=cp310-cp310
+ENV python=cp38-cp38
```
Comment (sighingnow) [inline docker/Dockerfile.vineyard-python-dev:18]: Any reason to downgrade that? It should be fine as it is in docker container.

File: docker/Dockerfile.vineyard-python-dev [hunk 2]
```diff
@@ -78,6 +78,8 @@ COPY modules/io /work/v6d/modules/io
 
 COPY python/vineyard /work/v6d/python/vineyard
 
+COPY vineyardctl /work/v6d/python/vineyard/vineyardctl
```
Comment (sighingnow) [inline docker/Dockerfile.vineyard-python-dev:81]: Where to copy from? Looks like you never built it in the dockerfile.

File: docker/Dockerfile.vineyard-python-dev [hunk 3]
```diff
@@ -98,7 +100,7 @@ RUN cd /work/v6d/build && \
     rm -rf /work/v6d
 
 # generate final image
-FROM python:3.10-slim-buster
+FROM python:3.8-slim-buster
```
Comment (sighingnow) [inline docker/Dockerfile.vineyard-python-dev:103]: As commented above, 3.10 should be preferred.

File: k8s/cmd/commands/create/create.go [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+// createCmd creates serveral vineyard jobs on kubernetes
+var createCmd = &cobra.Command{
+	Use:   "create",
+	Short: "Create the vineyard jobs on kubernetes",
+	Long: `Create a specific vineyard job on kubernetes.
+For example:
+
+# create the backup job on kubernetes
+vineyarctl -k /home/gsbot/.kube/config create backup --vineyardd-name vineyardd-sample --vineyardd-namespace vineyard-system`,
```
Comment (sighingnow) [inline k8s/cmd/commands/create/create.go:32]: No `-k`, use `--kubeconfig` or resolving from `KUBECONFIG` environments. Keep consistent with `kubectl`.

File: k8s/cmd/commands/create/create_backup.go [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+the current vineyard cluster quickly. For persistent storage, you could specify the pv spec and
+pv spec.
+
+For example:
+
+# create a backup job for the vineyard cluster on kubernetes
+vineyardctl create backup --vineyardd-name vineyardd-sample --vineyardd-namespace vineyard-system  \
+--limit 1000 --path /var/vineyard/dump  \
+--pv-spec '{"capacity": {"storage":"1Gi"}, "accessModes": ["ReadWriteOnce"], "storageClassName": "manual", "hostPath": {"path": "/var/vineyard/dump"}}'  \
+--pvc-spec '{"storageClassName": "manual", "accessModes": ["ReadWriteOnce"], "resources": {"requests": {"storage": "1Gi"}}}'`,
```
Comment (sighingnow) [inline k8s/cmd/commands/create/create_backup.go:47]: Looks not very pretty...

File: k8s/cmd/commands/delete/delete_vineyardd.go [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+
+# delete the default vineyardd cluster(vineyardd-sample) in the default namespace
+vineyardctl delete vineyardd
+
+# delete the specific vineyardd cluster in the vineyard-system namespace
+vineyardctl -n vineyard-system delete vineyardd --name vineyardd-test`,
+	Run: func(cmd *cobra.Command, args []string) {
+		if err := util.ValidateNoArgs("delete vineyardd", args); err != nil {
+			util.ErrLogger.Fatal("failed to validate delete vineyardd command args and flags: ", err,
+				"the extra args are: ", args)
```
Comment (sighingnow) [inline k8s/cmd/commands/delete/delete_vineyardd.go:48]: Looks quite adhoc. Does cobra has it own internal validation?

Comment (sighingnow) [reply k8s/cmd/commands/delete/delete_vineyardd.go:48]: See: https://github.com/spf13/cobra/blob/main/user_guide.md#positional-and-custom-arguments

File: k8s/cmd/commands/drydelete/drydelete_vineyardd.go [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+
+		scheme, err := util.GetClientgoScheme()
+		if err != nil {
+			util.ErrLogger.Fatal("failed to get client-go scheme: ", err)
+		}
+
+		kubeclient, err := util.GetKubeClient(scheme)
+		if err != nil {
+			util.ErrLogger.Fatal("failed to get kube client: ", err)
+		}
```
Comment (sighingnow) [inline k8s/cmd/commands/drydelete/drydelete_vineyardd.go:56]: Could you make a utility like `CreateKubeClient()` to avoid these duplicated lines?

**Review by sighingnow — COMMENTED** [review]
- No `dry{apply/delete}` etc.. You could adding a `--dry-run` flag and I think most of those command should share most of code snippets with the non-dry variant. There seems a lot of duplication.

- Don't upgrade to kuberntes 1.26. I think it is VERY non-common to keep upgrading the kuberentes cluster to latest version. Is there any blocking issue that v1.24 doesn't work?

- There're lots of copy-paste-with-slight-modification pattern, e.g., `parseManifests`....

- The python part is not OK. I was thinking generate the python bindings from cobra's help information other than reimplemented it again.

**Comment by sighingnow** [comment]
> The python part is not OK. I was thinking generate the python bindings from cobra's help information other than reimplemented it again.

Ping me when comments above addressed. I could help in the python part.

**Comment by sighingnow** [comment]
BTW I have re-targeted the pull request to the main branch.

**Pull request merged**